packages/esm-patient-orders-app/src/api/api.ts

```diff
@@ -67,13 +74,27 @@ export function createEmptyEncounter(
     );
     visitStartDate;
   }
+
+  const { items, postDataPrepFunctions }: OrderBasketStore = orderBasketStore.getState();
+  const patientItems = items[patientUuid];
+
+  const orders: Array<OrderPost> = [];
+  for (let grouping in patientItems) {
+    const groupOrders = patientItems[grouping];
+    for (let i = 0; i < groupOrders.length; i++) {
+      const order = groupOrders[i];
+      orders.push(postDataPrepFunctions[grouping](order, patientUuid, null));
+    }
+  }
+
```

Comment:
Instead of this `for loop`, you can use built-in react method `map`. Although functionality of both are almost same but `map` is faster than `for loop`

Reply:
Hi @raghavawasthi24!
I believe you mean to say `forEach`, since the `map` function is not a loop, but is used to map the item in an array to a custom value. `forEach` should be used if you don't want to return anything from the loop.